File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int max = nums.length / 2;
        
        HashSet<Integer> numSet = new HashSet<>();
        
        for (int i = 0; i < nums.length; i++) {
            numSet.add(nums[i]);
        }
        
        return numSet.size() > max ? max : numSet.size();
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <array>
#include <set>
using namespace std;

int solution(vector<int> nums)
{
    int max = nums.size() / 2;
    set<int> s;
    
    for (int i = 0; i < nums.size(); i++) {
        s.insert(nums[i]);
    }
    
    return s.size() > max ? max : s.size();

}